02week/pigLatin.js

```diff
@@ -7,22 +7,68 @@ const rl = readline.createInterface({
   output: process.stdout
 });
 
+//<*******************Whiteboard******************************>
+// 1. declare a GLOBAL variable to store the vowelIndex
+//
+// 2. create function isInputValid() that makes sure that the input does not have any spaces between letters or any numbers
+// IF the word contains no spaces or numbers (this is done using regEx search() for spaces and numbers)
+//r-eturn true
+// ELSE
+// -Throw an error that the input is invalid
+// 3. function findFirstVowel() searches the string for the index of the first vowel using regEx search() and assigns it to vowelIndex
+// 4. function pigLatin():
+//-force input to lower case and remove whitespaces
+//-call findFirstVowel() to get first vowel index
+// -if/else statement to determine if the first letter is a vowel:
+// IF the first letter in vowelIndex is a vowel (use regEx search(). if it's 0...
+// THEN
+// -return the full string plus ‘yay”)
+// ELSE
+// -Once you have the index of the first vowel (other than index 0), create a new variable newWord and assign the subString() of the original string to it
+// -return the newWord variable plus the original string starting at the index position of the vowel (from the variable vowelIndex) plus “ay”
+// ELSE
+// -return error that the word has no vowels
+//<****************************Begin code*********************************>
 
-function pigLatin(word) {
+let vowelIndex;
```

Comment:
This is unnecessary as a global variable, you could declare this with in pigLatin